### ISSUE TITLE
plugin and app engine

### ISSUE THREAD

**Issue by Pacane** [redstone-dart/redstone_web_socket#1]
Have you guys successfully deployed redstone with the web_socket_plugin to app engine managed vms?

I'got the the normal resources matching the end points, but it seems that on app engine (and even locally on gcloud) the requests never go to the methods

```
 @OnOpen()
  void onOpen(WebSocketSession session) {
    print("connection established");
  }

  @OnMessage()
  void onMessage(String message, WebSocketSession session) {
    print("message received: $message");
  }
```

This snippet works fine when I launch the server without gcloud. However, when I check the logs at startup I see that RedStone did its setup correctly (I suppose?)

`configured websocket handler for /ws: socket_handler.WebSocketEndPoint`

When querying the end point on my client I keep getting `failed: Error during WebSocket handshake: Unexpected response code: 404`

**Comment by cgarciae** [comment]
Can you post your `main`?

**Comment by Pacane** [comment]
Sure.. so I have a project, that I just want to wrap for the `runAppEngine` function from the appengine package.

So here's the main of my app-engine-wrapper project

``` dart
import 'package:appengine/appengine.dart';
import 'package:redstone/server.dart' as app;
import '../../pkg/server/bin/main.dart' as server;

void main() {
  logger.info("Running server's main");
  server.mainAppEngine().then((_) {
    runAppEngine((HttpRequest request) {
      app.handleRequest(request);
    }
  }).catchError((e) => logger.severe(e));
}
```

Here's the function that setups redstone:

``` dart
Future mainAppEngine() async {
  restPort = 4040;
  hostname = '0.0.0.0';

  await startGamesServer();
}

Future<HttpServer> startGamesServer() {
  logger.level = Level.FINE;
  setupLogging();

  app.addPlugin(getWebSocketPlugin());

  app.addModule(modules.getRestModule());

  return app.start(address: hostname, port: restPort).catchError(showError);
}
```

in `modules.getRestModule()` I bind all my resources + things that need to be injected + my class that hold `@WebSocketHandler("/ws")`

**Comment by cgarciae** [comment]
See redstone-dart/redstone#49

**Comment by Pacane** [comment]
Yeah, I used the code provided by @luizmineo in that issue, and it works fine for `@app.Group()`  and `@app.Route()` on AppEngine, but doesn't seem to work with the `@WebSocketHandler`

**Comment by Pacane** [comment]
I notices that the code provided by Luiz mentions that you have to _not call_ `app.start()`. I've tried that too, and doesn't seem to work better. (Instead of calling the API on `:4040` as in the sample above, I tried calling it with `:8080`).

**Comment by cgarciae** [comment]
You should call

``` dart
app.setup(); //<== Important
runAppEngine(app.handleRequest).catchError (...);
```

If that doesn't work, try a Redstone Hello World without websockets on App Engine just to make sure its really a problem of the plugin.
